modules/videoModule/index.js

```diff
@@ -0,0 +1,172 @@
+import { config } from '../../src/config.js';
+import { find } from '../../src/polyfill.js';
+import * as events from '../../src/events.js';
+import {
+  allVideoEvents, AUCTION_AD_LOAD_ATTEMPT, allVideoAuctionEvents,
+  AD_IMPRESSION, AD_ERROR, BID_VIDEO_IMPRESSION, BID_VIDEO_ERROR, allVideoBidEvents, AUCTION_AD_LOAD_ABORT
+} from './constants/events.js'
+import CONSTANTS from '../../src/constants.json';
+import { videoCoreFactory } from './coreVideo.js';
+import { coreAdServerFactory } from '../adServerModule/adServer.js';
+import { videoImpressionVerifierFactory } from './videoImpressionVerifier.js';
+import { mergeDeep } from '../../src/utils.js';
+import { getGlobal } from '../../src/prebidGlobal.js';
+
+/**
+ * This module adds User Video support to prebid.js
+ * @module modules/videoModule
+ */
+
+events.addEvents(allVideoEvents);
+events.addEvents(allVideoAuctionEvents);
+events.addEvents(allVideoBidEvents);
+
+export function PbVideo(videoCore_, getConfig_, pbGlobal_, pbEvents_, videoEvents_, adServerCore_, videoImpressionVerifierFactory_) {
+  const videoCore = videoCore_;
+  const getConfig = getConfig_;
+  const pbGlobal = pbGlobal_;
+  const requestBids = pbGlobal.requestBids;
+  const pbEvents = pbEvents_;
+  const videoEvents = videoEvents_;
+  const adServerCore = adServerCore_;
+  const videoImpressionVerifierFactory = videoImpressionVerifierFactory_;
+  let videoImpressionVerifier;
+
+  function init() {
+    const cache = getConfig('cache');
+    videoImpressionVerifier = videoImpressionVerifierFactory(!!cache);
+    getConfig('video', ({ video }) => {
+      video.providers.forEach(provider => {
+        videoCore.registerProvider(provider);
+        videoCore.onEvents(videoEvents, (type, payload) => {
+          pbEvents.emit(type, payload);
+        }, provider.divId);
+
+        const adServerConfig = provider.adServer;
+        if (adServerConfig) {
+          adServerCore.registerAdServer(adServerConfig);
+        }
+      });
+    });
+
+    requestBids.before(enrichAdUnits, 40);
```

Comment:
is the explicit priority set to `40` necessary? if so, can you add a comment explaining why?

Reply:
No reason, it seemed to be the common priority across modules. Not sure how to determine the right priority; any thoughts ?

Reply:
Several ask for priority 40 because they need to run after GDPR processing, which is set at priority 50. In your case I don't think it matters - but I haven't tested.

Reply:
I understand. The only requirement is for the function to run before **buildRequests** is called on the Bid Adapters, since the objective of the function is to write all of the ortb video and content info, for the bid adapters to consume before making a bid.